_posts/2022-10-13-access-the-k8s-api-from-functions.md

```diff
@@ -0,0 +1,236 @@
+subjects:
+  - kind: ServiceAccount
+    name: fn-events
+    namespace: openfaas-fn
+---
+```
+
+Functions can only assume a ServiceAccount in the namespace in which they are deployed. The `fn-events` ServiceAccount is created in the `openfaas-fn` namespace. This is the default namespace used for OpenFaaS functions.
+
+> Learn more about [RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) and [Service Accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) in the Kubernetes documentation.
+
+You can assign a ServiceAccount to a function by adding the annotation `com.openfaas.serviceaccount` in the functions `stack.yaml` file. 
+
+```yaml
+functions:
+  get-events:
+    lang: golang-middleware
+    handler: ./get-events
+    image: welteki2/get-events:0.1.1
+    annotations:
+      com.openfaas.serviceaccount: fn-events
+```
+
+Deploy the function and invoke it with `curl`:
+
+```bash
+export OPENFAAS_URL="http://127.0.0.1:8080"
+curl -i $OPENFAAS_URL/function/get-events
+
+HTTP/1.1 200 OK
+Content-Length: 798
+Content-Type: text/plain; charset=utf-8
+Date: Thu, 13 Oct 2022 13:41:09 GMT
+X-Call-Id: b1af2c1b-bd06-4b9c-969a-a7d8e065a2e8
+X-Duration-Seconds: 0.005199
+X-Start-Time: 1665668469270352257
+
+LAST SEEN           TYPE   REASON OBJECT                      MESSAGE
+13 Oct 22 13:40 UTC Normal Synced Function/bcrypt             Function synced successfully
+13 Oct 22 13:40 UTC Normal Synced Function/cows               Function synced successfully
+13 Oct 22 13:40 UTC Normal Synced Function/delay-sleep        Function synced successfully
+13 Oct 22 13:40 UTC Normal Synced Function/health             Function synced successfully
+13 Oct 22 13:40 UTC Normal Synced Function/list-of-namespaces Function synced successfully
+13 Oct 22 13:40 UTC Normal Synced Function/sleep              Function synced successfully
+13 Oct 22 13:40 UTC Normal Synced Function/ready              Function synced successfully
+13 Oct 22 13:40 UTC Normal Synced Function/get-events         Function synced successfully
+```
+
+## Trigger workloads on a schedule
+Depending on the workload you are running in your function you might want to trigger it on a timed-basis. OpenFaaS has support for this through the [cron event-connector](https://docs.openfaas.com/reference/cron/#kubernetes).
+
+You can deploy the cron-connector with [arkade](https://github.com/alexellis/arkade).
+```
+arkade install cron-connector
+```
+
+Two annotation are required to trigger the function, a `topic` of `cron-function` and a `schedule` using a valid Cron expression.
+
+Add these annotations to the `stack.yml`:
+
+```yaml
+functions:
+  get-events:
+    lang: golang-middleware
+    handler: ./get-events
+    image: welteki2/get-events:0.1.1
+    annotations:
+      com.openfaas.serviceaccount: fn-events
+      topic: cron-function
+      schedule: "* */1 * * *"
+```
+
+This will trigger the function every hour.
+
+
+# Conclusion
```

Comment:
This is very sudden.

I like to see more of a "what if" here.

That's where you get the imagination going with some practical suggestions and/or tasks someone may use this for.

Also, links to other similar posts or tutorials / docs links.